app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationFetcher.kt

```diff
@@ -53,87 +54,63 @@ class NotificationFetcher @Inject constructor(
-                    // when it is released.
+                    notificationsByType.forEach { notificationsGroup: Map.Entry<Notification.Type, List<Notification>> ->
+                        val newNotifications = ArrayList<NotificationManagerCompat.NotificationWithIdAndTag>();
 
-                    notificationsByType.forEach { notificationsGroup ->
-                        notificationsGroup.value.forEach { notification ->
-                            val androidNotification = NotificationHelper.make(
+                        // NOTE: Enqueue the summary first: Only this way one can avoid running into notification rate limits;
+                        //   which also would prohibit the group summary to be shown and thus proper grouping!
+                        newNotifications.add(
+                            NotificationHelper.makeSummaryNotification(
                                 context,
                                 notificationManager,
-                                notification,
                                 account,
-                                notificationsGroup.value.size == 1
-                            )
-                            notificationManager.notify(
-                                notification.id,
-                                account.id.toInt(),
-                                androidNotification
+                                notificationsGroup.key,
+                                notificationsGroup.value
                             )
+                        )
 
-                            // Android will rate limit / drop notifications if they're posted too
-                            // quickly. There is no indication to the user that this happened.
-                            // See https://github.com/tuskyapp/Tusky/pull/3626#discussion_r1192963664
-                            delay(1000.milliseconds)
+                        notificationsGroup.value.forEach { notification ->
+                            newNotifications.add(
+                                NotificationHelper.make(
+                                    context,
+                                    notificationManager,
+                                    notification,
+                                    account
+                                )
+                            )
                         }
-                    }
 
-                    NotificationHelper.updateSummaryNotifications(
-                        context,
-                        notificationManager,
-                        account
-                    )
+                        notificationManagerCompat.notify(newNotifications)
+
+                        // NOTE this can schedule multiple (summary) notifications in short succession (normally 1-3).
+                        //   They can "collapse" to only one audible one if fast enough.
```

Comment:
Could this be outside of the `notificationsByType.forEach` loop as well?

Reply:
Done.